components/service/experiments/src/main/java/mozilla/components/service/experiments/ExperimentsUpdater.kt

```diff
@@ -84,3 +123,23 @@ internal class ExperimentsUpdater(
         private const val EXPERIMENTS_COLLECTION_NAME = "focus-experiments"
     }
 }
+
+/**
+ * This class represents a [Worker] object that periodically updates the experiments.
+ * @param updater used so that the updater can be injected from tests.
+ */
+internal class ExperimentsUpdaterWorker(
+    context: Context,
+    workerParameters: WorkerParameters,
+    @VisibleForTesting(otherwise = VisibleForTesting.PRIVATE)
+    internal var updater: ExperimentsUpdater = Experiments.updater
```

Comment:
I didn't intend for it to get passed under normal circumstances, it should resort to the default of using the singleton's updater.  I added this in order to inject an `ExperimentsUpdater` from a test so that I could initialize it and manipulate it when unit testing the `ExperimentsUpdateWorker` class.

Reply:
It looks like since I've added this, my manual test in samples-glean has broken because of it.  Looks like I'll have to find another way around this...